reconcile/utils/vault.py

```diff
@@ -211,6 +201,7 @@ def write(self, secret):
             self._write_v2(secret_path, data)
 
     def _write_v2(self, path, data):
+        # do not forget to run `self._read_all_v2.cache_clear()` if this ever get's implemented
         raise NotImplementedError('vault_client write v2')
 
     def _write_v1(self, path, data):
```

Comment:
should we `cache_clear` here?

Reply:
if we cache the result of `_read_all_v2` we should propably also invalidate this if we touch a secret in a v2 engine